4iprocessing/generate_initial_config.py

```diff
@@ -0,0 +1,130 @@
+                czi_list = [x for x in os.listdir(ppath) if ('.czi' in x)&bool(re.search('20x',x,re.IGNORECASE))]
+                # print(czi_list)
+                # subd = {}
+                # subd['name'] = round_num
+                # subd['details'] = []
+
+                for czi_name in czi_list:
+                    fpath = os.path.join(ppath,czi_name)
+                    fpathr = fpath.replace(os.sep,'/')
+                    fpath2 ='"'+fpathr+'"'
+                    fpath2 = fpathr
+                    fpath = fpath2.replace("'",'')
+
+                    # each round of imaging is a multi-scene file like with dimensions STCZYX (only "time lapse rounds" have T>1). All rounds have S>1 and T>1
+                    # define the location of the reference channel in the list of channels
+                    # ideally the reference channel will be the image containing the nuclear fluorescence (or perhaps brightfield)
+                    # the reference image type (e.g. nuclear dye or brightfield) should be the consisently chosen for all rounds
+                    # if the image is a timelapse or round01 then choose the reference channel to be the last channel in the image set , =-1.
+
+                    ref_channel = -2
+                    if bool(re.search('time|1(?![0-9])',round_num, re.IGNORECASE)) and round_num!="Round 11": 
+                        #print('TODO: make sure this doesnt capture round 11')
+                        print("round num is {}".format(round_num))
+                        ref_channel=-1
+                        print("ref channel is {}".format(ref_channel))
+
+                    # use the parent czi file for metadata
+                    # find the channel names for the image file
+                    reader = AICSImage(fpath)
+                    channels = reader.channel_names
+                    
+                    # iterate through all scenes in metadata and look for valid scenes (i.e. scenes with image data)
+                    # if scene lacks image data, mark it as a scene to toss (it has no image data!)
+                    scenes_to_toss=[]
+                    for scene in reader.scenes:
+                        reader.set_scene(scene)
+                        si =reader.current_scene_index
+                        try: 
+                            dims = reader.dims
+                        except:
+                            scenes_to_toss.append(si+1)
+
+                    zscenes_to_toss=','.join([str(x) for x in scenes_to_toss])
+                    zchannels = ','.join(channels)
+                    #the names for these yaml entries are weird (i.e. "iround") because they get organized alphabetically and I want them to appear in a desired order
+                    print(f"channels are {zchannels}")
+                    detailid={}
+                    detailid['round'] = round_num
+                    detailid['item'] = 'czi'
+                    detailid['zchannels'] = zchannels
```

Comment:
change name to `channels` instead of `zchannels`

Reply:
I also want channels to be at the bottom, so maybe move this line below 107.

Reply:
ohh, nice catch